Source/objects.cpp

```diff
@@ -2684,9 +2683,8 @@ bool OperateShrineGloomy(int pnum)
 		case ItemType::Bow:
 		case ItemType::Mace:
 		case ItemType::Staff:
-			item._iMaxDam--;
-			if (item._iMaxDam < item._iMinDam)
-				item._iMaxDam = item._iMinDam;
+			if (!item._iDamage.IsFixed())
+				item._iDamage -= { 0, 1 };

```

Comment:
It changes the logic to avoid "setting a value and undoing the operation".

Previously, this was decrementing max damage, and if it became less than min damage (only possible if the values were equal prior to the decrement, or "fixed") it would set maxDamage to minDamage, basically undoing the operation.

The new logic is more sane and does the same thing: if the damage is already fixed, there is no need decrementing max component and then undoing it. If it is _not_ fixed, then it will always be able to decrement max damage without creating the problem.